R/input-utils.R

```diff
@@ -90,31 +90,39 @@ generateOptions <- function(inputId, selected, inline, type = 'checkbox',
   div(class = "shiny-options-group", options)
 }
 
+# If a list/vector is unnamed, give it blank names
+ensureNamed <- function(x) {
+  if (is.null(names(x))) names(x) <- character(length(x))
+  x
+}
+
+# Take a vector or list, and convert to list. Also, if any children are
+# vectors with length > 1, convert those to list. If the list is unnamed,
+# convert it to a named list with blank names.
+listify <- function(x) UseMethod("listify", x)
```

Comment:
My gut reaction is this should be rewritten to be a `if` / `else`.  

* It wouldn't need to be exported to work.
* It could handle odd s3 cases better (things that don't have atomic classes).

Setup: 
```
if `is.list`, `is.null`
  "code for `is.list`"
else if `is.numeric`, `is.complex`, `is.logical`, `is.factor`
  "code for `is.numeric`"
else 
  "code for `is.character`"
```

Reply:
You know, now that it has all the `@export` noise, I have to agree. I swear, it was looking pretty tight before all that 😄

Reply:
Since the `listify` generic isn't exported from the package, you don't need to `@export` every method, and therefore nothing will be added to the NAMESPACE file. That said, I also lean toward having a single function with if-else here.